hacep/src/main/java/org/kie/hacep/core/infra/utils/SnapshotOnDemandUtils.java

```diff
@@ -70,39 +66,32 @@ public static SnapshotInfos askASnapshotOnDemand(EnvConfig config,
         }
     }
 
-    private static SnapshotInfos buildNewSnapshotOnDemand(EnvConfig config,
-                                                          LocalDateTime limitAge) {
-        SnapshotMessage snapshotMsg = askAndReadSnapshotOnDemand(config,
-                                                                 limitAge);
+    private static SnapshotInfos buildNewSnapshotOnDemand(EnvConfig envConfig, LocalDateTime limitAge) {
+        SnapshotMessage snapshotMsg = askAndReadSnapshotOnDemand(envConfig, limitAge);
         KieSession kSession = null;
+        KieContainer kieContainer = null;
         try (ByteArrayInputStream in = new ByteArrayInputStream(snapshotMsg.getSerializedSession())) {
-            KieSessionConfiguration conf = KieServices.get().newKieSessionConfiguration();
-            conf.setOption(ClockTypeOption.get("pseudo"));
-            kSession = KieServices.get().getMarshallers().newMarshaller(getKieContainer().getKieBase()).unmarshall(in,
-                                                                                                                   conf,
-                                                                                                                   null);
+            KieServices ks = KieServices.get();
+            if(ks != null) {
+                KieSessionConfiguration conf = ks.newKieSessionConfiguration();
+                conf.setOption(ClockTypeOption.get("pseudo"));
+                String[] partsSerialized = snapshotMsg.getKjarGAV().split(":");
```

Comment:
This could be null if the system is running in a not updatable setup

Reply:
Yep, I'll centralize the code used in the DroolsConsumerHandler (with handling of the updatable or not) and the code here.